### PR TITLE
observer benchmarking

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29508 observer benchmarking**

Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

### CODE REVIEW

File: benchmarks/operator_benchmark/pt/observer_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+
+observer_configs_short = op_bench.config_list(
+    # mode is used to show the direction of the benchmark:
+    # if 'Q', benchmark quantization, else dequantization
+    attr_names=['C', 'M', 'N', 'dtype', 'qscheme'],
+    attrs=[
+        [3, 512, 512, torch.quint8, torch.per_tensor_affine],
+    ],
+
```
Comment (mingzhe09088) [inline benchmarks/operator_benchmark/pt/observer_test.py]: remove this line?

**Comment by mingzhe09088** [comment]
Can you rename the test name to qobserver_test.py so it is consistent with other quantized tests?

File: benchmarks/operator_benchmark/pt/qobserver_test.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+    tags=['short']
+)
+
+qobserver_configs_long = op_bench.cross_product_configs(
+    C=[1, 3, 8],
+    M=[256, 1024],
+    N=[256, 1024],
+    dtype=[torch.quint8],  # dtype doesn't change the timing
+    qscheme=[torch.per_tensor_affine, torch.per_tensor_symmetric,
```
Comment (raghuramank100) [inline benchmarks/operator_benchmark/pt/qobserver_test.py]: For observers, cross products wont work.  per-channel qconfig only works with per-channel observer for example. I would write this by having one set of configs for per-tensor-quant (HistogramObserver/MinMax/MovingAverage) and another set of configs for per-channel quant (PerChannelMinMax/PerChannelMovingAverage)
We can fix the dtype to be torch.quint8 for all these tests.

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
We will need to restrict configs so that the tests run, thanks

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**